internal/component/pyroscope/write/write.go

```diff
@@ -390,59 +403,96 @@ func (f *fanOutClient) AppendIngest(ctx context.Context, profile *pyroscope.Inco
 
 	// Send to each endpoint concurrently
 	for endpointIdx, endpoint := range f.config.Endpoints {
+		var (
+			endpoint = endpoint
+			i        = endpointIdx
+			backoff  = backoff.New(ctx, backoff.Config{
+				MinBackoff: f.config.Endpoints[i].MinBackoff,
+				MaxBackoff: f.config.Endpoints[i].MaxBackoff,
+				MaxRetries: f.config.Endpoints[i].MaxBackoffRetries,
+			})
+			err error
+		)
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			u, err := url.Parse(endpoint.URL)
-			if err != nil {
-				util.ErrorsJoinConcurrent(&errs, fmt.Errorf("parse URL for endpoint[%d]: %w", endpointIdx, err), &errorMut)
-				return
-			}
 
-			u.Path = path.Join(u.Path, profile.URL.Path)
+			for {
+				err = func() error {
+					u, err := url.Parse(endpoint.URL)
+					if err != nil {
+						return fmt.Errorf("parse URL for endpoint[%d]: %w", i, err)
+					}
 
-			// attch labels
-			u.RawQuery = query.Encode()
+					u.Path = path.Join(u.Path, profile.URL.Path)
 
-			req, err := http.NewRequestWithContext(ctx, "POST", u.String(), bytes.NewReader(profile.RawBody))
-			if err != nil {
-				util.ErrorsJoinConcurrent(&errs, fmt.Errorf("create request for endpoint[%d]: %w", endpointIdx, err), &errorMut)
-				return
-			}
+					// attach labels
```

Comment:
Nit: Regarding labels, I observed we have a [slightly different validation](https://github.com/grafana/alloy/blob/cd6cce5f9eaeb4687d14f989fe07d379b08e2577/internal/component/pyroscope/write/write.go#L314) in `Append` and `AppendIngest`. In `Append` we're skipping labels starting with `__` while we don't check it in `AppendIngest`.

Feel free to omit this; we could do it in a separate PR.

Reply:
do we have an issue for this? let's do as a separate change

Reply:
ok, I've just [created](https://github.com/grafana/alloy/issues/4225) it to not forget about it. Not sure whether I'll be able to work on it this week, otherwise I can do it once I'm back 🙂

Reply:
I guess theres no urgency. 
I am only fixing the metrics here because I was trying to reproduce an issue and was not receiving any metrics and it stood in my way